### PR TITLE
r.grow.distance: add minimum and maximum distance options

### PR DESCRIPTION
**Pull request by metzm**
add maximum distance option

### CODE REVIEW

File: raster/r.grow.distance/main.c [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * AUTHOR(S):    Marjorie Larson - CERL
  *               Glynn Clements
  *
- * PURPOSE:      Generates a raster map layer with contiguous areas 
+ * PURPOSE:      Generates a raster map layer with contiguous areas
  *               grown by one cell.
  *
  * COPYRIGHT:    (C) 2006-2013 by the GRASS Development Team
```
Comment (neteler) [inline raster/r.grow.distance/main.c:12]:
```suggestion
 * PURPOSE:      Generates a raster map layer with contiguous areas
 *               grown by one cell.
 *
 * COPYRIGHT:    (C) 2006-2021 by the GRASS Development Team
```

File: raster/r.grow.distance/main.c [hunk 2]
```diff
@@ -190,7 +203,41 @@ int main(int argc, char **argv)
...
+	if (sscanf(opt.max->answer, "%lf", &maxdist) != 1) {
+	    G_warning(_("Invalid %s value '%s', ignoring."),
+		      opt.max->key, opt.max->answer);
+
+	    maxdist = -1;
+	}
+    }
+
+    if (mindist > 0 && maxdist > 0 && mindist >= maxdist) {
+	G_fatal_error(_("%s must be smaller than %s."),
```
Comment (marisn) [inline raster/r.grow.distance/main.c:228]: Add a translator comment before this message.

File: raster/r.grow.distance/r.grow.distance.html [hunk 3]
```diff
@@ -64,6 +64,15 @@ <h2>NOTES</h2>
...
 
+<p>
+If <b>minimum_distance</b> is given, all cells with a distance smaller
+than <b>minimum_distance</b> will be set to NULL.
+
+<p>
+If <b>maximum_distance</b> is given, all cells with a distance larger
+than <b>maximum_distance</b> will be set to NULL. The resultant output
+is equivalent to a buffer.
+
```
Comment (marisn) [inline raster/r.grow.distance/r.grow.distance.html:75]: Add an explanation what will be the outcome if both – min and max distance are given. Will it be a ring (doughnut)?

**Review by marisn — APPROVED** [review]

File: raster/r.grow.distance/main.c (hunk 2 above)
Comment (metzm) [reply raster/r.grow.distance/main.c:228]: I don't know what you mean with "translator comment", please explain.

File: raster/r.grow.distance/r.grow.distance.html (hunk 3 above)
Comment (metzm) [reply raster/r.grow.distance/r.grow.distance.html:75]: I have updated the manual to explain the result if both min and max distance restrictions are given.

File: raster/r.grow.distance/main.c (hunk 2 above)
Comment (nilason) [reply raster/r.grow.distance/main.c:228]: I suppose what’s described in https://trac.osgeo.org/grass/wiki/Submitting/C#Messages.

Comment (metzm) [reply raster/r.grow.distance/main.c]: Tried to add a translator comment, ok @marisn ?

Comment (marisn) [reply raster/r.grow.distance/main.c]: Seems good. Sometimes simply it is hard to understand how to translate a phrase when it is unknown what %s stands for in the final message.

**Pull request merged**